nethttp/server.go

```diff
@@ -75,6 +75,10 @@ func MWSpanObserver(f func(span opentracing.Span, r *http.Request)) MWOption {
 //		}),
 //   )
 func Middleware(tr opentracing.Tracer, h http.Handler, options ...MWOption) http.Handler {
+	return MiddlewareFunc(tr, h.ServeHTTP, options...)
+}
+
+func MiddlewareFunc(tr opentracing.Tracer, h func(http.ResponseWriter, *http.Request), options ...MWOption) http.Handler {
```

Comment:
This can be easily composed using stdlib, like `Middleware(tr, http.HandlerFunc(h), ...)`.

But given that this is a fairly common pattern in many libs, I don't see strong reason against providing this.

Reply:
I was unaware of the `HandlerFunc` type. I'm replacing that in the method signature now.

Reply:
Hah, `HandlerFunc` is literally right there in the return value. 🤦‍♂️